### ISSUE TITLE
Animation extension

### ISSUE THREAD

**Issue by jamesplease** [jamesplease/puppets#18]
Animations will be based on css className.

``` html
<span class='my-list'></span>
```

At certain points in the Application, Puppets will add additional classNames to the element in an attempt to see if it begins to animate. For instance, `p-enter` will be added when a view is being shown, and `p-exit` for a view that is being exited.

Puppets will intelligently handle the situation that the animation has not been configured, so it should be a pretty seamless experience for developers.

Javascript animations would be registered via:

``` js
Puppets.registerAnimation('.my-list', {
  enter(element, done) {},
  exit(element, done) {}
});
```

Situations that trigger animations:
- Showing a child view
- Removing a child view
- Swapping views
- In other words, anything that happens through a Region
- Routing actions

Puppets.Animate will be a separate library.

\* this is based off Angular's implementation of animations, which I think is pretty baller
